### ISSUE TITLE
RuntimeError: latex was not able to process the following string: '_auto'

### ISSUE THREAD

**Issue by lupin0lin** [gwpy/gwpy#789]
To whom will concern my question,

I met a problem to perform the python code built by LIGO team on the jupyter notebook. When I tried to perform the following simple code without the environment of the jupyter notebook, I can obtain the correct output and the figure (https://i.stack.imgur.com/Bz5zO.png).

+++++++++++++++++++++++++++++++++
```
from gwpy.timeseries import TimeSeries
from numpy import random
series = TimeSeries(random.random(1000), sample_rate=100, unit='m')
plot = series.plot()
plot.show()
```
+++++++++++++++++++++++++++

But when I applied the following similar python code in the jupyter notebook:

++++++++++++++++
```
%matplotlib inline

%config InlineBackend.figure_format = 'retina'
from gwpy.timeseries import TimeSeries
from numpy import random
series = TimeSeries(random.random(1000), sample_rate=100, unit='m')
plot = series.plot()
```
++++++++++++++++

I obtained the runtime error related to the latex. Since I can get the correct output if I do not use the jupyter, I speculate that the problem is caused by the transformation of the plot to present on the notebook environment. I tried to check the related runtime error related to latex and install the packages (e.g., dvipng, texlive-latex-extra & texlive-fonts-recommended) through Macports or re-install MacTeX in my machine, but the problem still exists. Here comes the warning & error messages I obtained on the jupyter notebook.

---------------------------------------------------------------------------
```
RuntimeError                              Traceback (most recent call last)
/Users/lupin/Library/Python/2.7/lib/python/site-packages/IPython/core/formatters.pyc in __call__(self, obj)
    332                 pass
    333             else:
--> 334                 return printer(obj)
    335             # Finally look for special method names
    336             method = get_real_method(obj, self.print_method)

/Users/lupin/Library/Python/2.7/lib/python/site-packages/IPython/core/pylabtools.pyc in <lambda>(fig)
    247         png_formatter.for_type(Figure, lambda fig: print_figure(fig, 'png', **kwargs))
    248     if 'retina' in formats or 'png2x' in formats:
--> 249         png_formatter.for_type(Figure, lambda fig: retina_figure(fig, **kwargs))
    250     if 'jpg' in formats or 'jpeg' in formats:
    251         jpg_formatter.for_type(Figure, lambda fig: print_figure(fig, 'jpg', **kwargs))

/Users/lupin/Library/Python/2.7/lib/python/site-packages/IPython/core/pylabtools.pyc in retina_figure(fig, **kwargs)
    137 def retina_figure(fig, **kwargs):
    138     """format a figure as a pixel-doubled (retina) PNG"""
--> 139     pngdata = print_figure(fig, fmt='retina', **kwargs)
    140     # Make sure that retina_figure acts just like print_figure and returns
    141     # None when the figure is empty.

/Users/lupin/Library/Python/2.7/lib/python/site-packages/IPython/core/pylabtools.pyc in print_figure(fig, fmt, bbox_inches, **kwargs)
    129 
    130     bytes_io = BytesIO()
--> 131     fig.canvas.print_figure(bytes_io, **kw)
    132     data = bytes_io.getvalue()
    133     if fmt == 'svg':

/Library/Python/2.7/site-packages/matplotlib/backend_bases.pyc in print_figure(self, filename, dpi, facecolor, edgecolor, orientation, format, **kwargs)
   2212                     **kwargs)
   2213                 renderer = self.figure._cachedRenderer
-> 2214                 bbox_inches = self.figure.get_tightbbox(renderer)
   2215 
   2216                 bbox_artists = kwargs.pop("bbox_extra_artists", None)

/Library/Python/2.7/site-packages/matplotlib/figure.pyc in get_tightbbox(self, renderer)
   2188         for ax in self.axes:
   2189             if ax.get_visible():
-> 2190                 bb.append(ax.get_tightbbox(renderer))
   2191 
   2192         if len(bb) == 0:

/Library/Python/2.7/site-packages/matplotlib/axes/_base.pyc in get_tightbbox(self, renderer, call_axes_locator)
   4168             bb.append(self._right_title.get_window_extent(renderer))
   4169 
-> 4170         bb_xaxis = self.xaxis.get_tightbbox(renderer)
   4171         if bb_xaxis:
   4172             bb.append(bb_xaxis)

/Library/Python/2.7/site-packages/matplotlib/axis.pyc in get_tightbbox(self, renderer)
   1158         for a in [self.label, self.offsetText]:
   1159             if a.get_visible():
-> 1160                 bb.append(a.get_window_extent(renderer))
   1161 
   1162         bb.extend(ticklabelBoxes)

/Library/Python/2.7/site-packages/matplotlib/text.pyc in get_window_extent(self, renderer, dpi)
    920             raise RuntimeError('Cannot get window extent w/o renderer')
    921 
--> 922         bbox, info, descent = self._get_layout(self._renderer)
    923         x, y = self.get_unitless_position()
    924         x, y = self.get_transform().transform_point((x, y))

/Library/Python/2.7/site-packages/matplotlib/text.pyc in _get_layout(self, renderer)
    307                 w, h, d = renderer.get_text_width_height_descent(clean_line,
    308                                                         self._fontproperties,
--> 309                                                         ismath=ismath)
    310             else:
    311                 w, h, d = 0, 0, 0

/Library/Python/2.7/site-packages/matplotlib/backends/backend_agg.pyc in get_text_width_height_descent(self, s, prop, ismath)
    230             fontsize = prop.get_size_in_points()
    231             w, h, d = texmanager.get_text_width_height_descent(
--> 232                 s, fontsize, renderer=self)
    233             return w, h, d
    234 

/Library/Python/2.7/site-packages/matplotlib/texmanager.pyc in get_text_width_height_descent(self, tex, fontsize, renderer)
    499         else:
    500             # use dviread. It sometimes returns a wrong descent.
--> 501             dvifile = self.make_dvi(tex, fontsize)
    502             with dviread.Dvi(dvifile, 72 * dpi_fraction) as dvi:
    503                 page = next(iter(dvi))

/Library/Python/2.7/site-packages/matplotlib/texmanager.pyc in make_dvi(self, tex, fontsize)
    363                 self._run_checked_subprocess(
    364                     ["latex", "-interaction=nonstopmode", "--halt-on-error",
--> 365                      texfile], tex)
    366             for fname in glob.glob(basefile + '*'):
    367                 if not fname.endswith(('dvi', 'tex')):

/Library/Python/2.7/site-packages/matplotlib/texmanager.pyc in _run_checked_subprocess(self, command, tex)
    342                     prog=command[0],
    343                     tex=tex.encode('unicode_escape'),
--> 344                     exc=exc.output.decode('utf-8')))
    345         _log.debug(report)
    346         return report

RuntimeError: latex was not able to process the following string:
'_auto'

Here is the full report generated by latex:
This is pdfTeX, Version 3.14159265-2.6-1.40.18 (TeX Live 2017/MacPorts 2017_4) (preloaded format=latex)
 restricted \write18 enabled.
entering extended mode
(/Users/lupin/.matplotlib/tex.cache/1de80ee53f095837776b678f34112ba4.tex
LaTeX2e <2017-04-15>
Babel <3.10> and hyphenation patterns for 3 language(s) loaded.
(/opt/local/share/texmf-texlive/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(/opt/local/share/texmf-texlive/tex/latex/base/size10.clo))
(/opt/local/share/texmf-texlive/tex/latex/type1cm/type1cm.sty)
(/opt/local/share/texmf-texlive/tex/latex/base/textcomp.sty
(/opt/local/share/texmf-texlive/tex/latex/base/ts1enc.def))
(/opt/local/share/texmf-texlive/tex/latex/geometry/geometry.sty
(/opt/local/share/texmf-texlive/tex/latex/graphics/keyval.sty)
(/opt/local/share/texmf-texlive/tex/generic/oberdiek/ifpdf.sty)
(/opt/local/share/texmf-texlive/tex/generic/oberdiek/ifvtex.sty)
(/opt/local/share/texmf-texlive/tex/generic/ifxetex/ifxetex.sty)

Package geometry Warning: Over-specification in `h'-direction.
    `width' (5058.9pt) is ignored.


Package geometry Warning: Over-specification in `v'-direction.
    `height' (5058.9pt) is ignored.

) (./1de80ee53f095837776b678f34112ba4.aux)
(/opt/local/share/texmf-texlive/tex/latex/base/ts1cmr.fd)
*geometry* driver: auto-detecting
*geometry* detected driver: dvips
! Missing $ inserted.
<inserted text> 
                $
l.13 \fontsize{20.000000}{25.000000}{\rmfamily _
                                                auto}
No pages of output.
Transcript written on 1de80ee53f095837776b678f34112ba4.log.




<TimeSeriesPlot size 864x432 with 1 Axes>
```
---------------------------------------------------------------------------

Can anyone provide some suggestions to solve this issue?

Sincerely,

**Comment by duncanmmacleod** [comment]
@lupin0lin, can you post the version of gwpy you were using when this bug occurred? I'm pretty sure this was either fixes in 0.10.0 (the most recent release) or will be fixed in 0.11.0 (the next release).

**Comment by lupin0lin** [comment]
Dear Duncan Macleod,

I installed 'gwpy==0.8.1', 'matplotlib==2.2.2' in my own Macbookpro.
I also checked the environment of Docker containers providing by LIGO team:

```
%matplotlib inline
import matplotlib
print(matplotlib.__version__)
print(matplotlib.__file__)
import gwpy
print(gwpy.__version__)
print(gwpy.__file__)

2.1.2
/home/pycbc/pycbc-software/lib/python2.7/site-packages/matplotlib/__init__.pyc
0.8.1
/home/pycbc/pycbc-software/lib/python2.7/site-packages/gwpy/__init__.pyc
```

And it can correctly generate the output on the jupyter Notebook.
So should I really need to update my gwpy package?

**Comment by lupin0lin** [comment]
My problem has already been solved.
Because I installed the Jupyter with the python 2.7, installation of the "matplotlib" with the latest version will cause such a problem.
To solve the bug, I downgrade the matplotlib to 1.5.3 and it works well now.

**Comment by duncanmmacleod** [comment]
@lupin0lin, I'm a little confused about the status of this ticket. Are you saying the problem is resolved now, or is still present when using matplotlib>=2?

**Comment by lupin0lin** [comment]
Dear Duncan,
 
I solved the problem after downgrading the version of matplotlib (with v1.5.3).
If I still use the matplotlib >=2, I think the problem still exists.
However, I installed the jupyter notebook with Python v.2.7.10, which is the old python version.
I heard that the problem may also be solved (without downgrading the matplotlib) if I install the jupyter notebook with Python >=3.0.
In my case, I finally decided a simpler way to solve it.
